src/Utilities/IsNan.cpp

```diff
@@ -0,0 +1,22 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Utilities/IsNan.hpp"
+
+#include <cmath>
+
+#include "ErrorHandling/FloatingPointExceptions.hpp"
+
+bool is_nan(const double arg) noexcept {
```

Comment:
The standard has already claimed `::isnan` as a possible synonym for `std::isnan`.

Reply:
ok, maybe `isnan_safe`? I'm not asking you to make a change, just wondering if it's better to name this more descriptively to avoid confusion with the std version. Your decision.